habitat_sim/utils/viz_utils.py

```diff
@@ -26,6 +31,260 @@ def is_notebook():
+              idx 1 : whether camera is depth, semantic or color
+              idx 2 : whether camera is 1st person or 3rd person
+        "height" : camera height (same for all cameras)
+        "width" : camera width (same for all cameras)
+        "1st_POV_Loc" : location of 1st person camera (x,y,z)
+        "1st_POV_Orient" : rotation of 1st person camera around x,y,z axis
+        "3rd_POV_Loc" : location of 3rd person camera (x,y,z)
+        "3rd_POV_Orient" : rotation of 3rd person camera around x,y,z axis
+
+    Returns
+    -------
+    cfg : habitat_sim.Configuration
+        Configuration required to instance necessary Habitat components.
+
+    """
+    # simulator configuration
+    backend_cfg = habitat_sim.SimulatorConfiguration()
+    backend_cfg.scene.id = settings["scene"]
+    assert os.path.exists(backend_cfg.scene.id)
+    backend_cfg.enable_physics = settings["enable_physics"]
+
+    # sensor configurations
+    camera_list = settings["camera_list"]
+    # sensor configurations
+    # Note: all sensors must have the same resolution
+    sensors_config_dict = {}
+    for camera in camera_list:
+        camera_name = camera[0]
+        temp_dict = {}
+        camera_type = camera[1].lower()
+        # check if depth, semantic, or color
+        if "depth" in camera_type:
+            temp_dict["sensor_type"] = habitat_sim.SensorType.DEPTH
+        elif "semantic" in camera_type:
+            temp_dict["sensor_type"] = habitat_sim.SensorType.SEMANTIC
+        else:
+            temp_dict["sensor_type"] = habitat_sim.SensorType.COLOR
+
+        # resolution should be height x width
+        temp_dict["resolution"] = [settings["height"], settings["width"]]
+        # check if 1st or 3rd person camera
+        if ("firstperson" in camera[2].lower()) or ("1st" in camera[2].lower()):
+            temp_dict["position"] = settings["1st_POV_Loc"]  # [0.0, 0.6, 0.0]
+            # [0.0, 0.0, 0.0]
+            temp_dict["orientation"] = settings["1st_POV_Orient"]
+        else:
+            temp_dict["position"] = settings["3rd_POV_Loc"]  # [0.0, 1.0, 0.3]
+            # [-45, 0.0, 0.0]
+            temp_dict["orientation"] = settings["3rd_POV_Orient"]
+        sensors_config_dict[camera_name] = temp_dict
+
+    # sensor configurations
+    sensor_specs = []
+    for sensor_uuid, sensor_params in sensors_config_dict.items():
+        sensor_spec = habitat_sim.SensorSpec()
+        sensor_spec.uuid = sensor_uuid
+        sensor_spec.sensor_type = sensor_params["sensor_type"]
+        sensor_spec.resolution = sensor_params["resolution"]
+        sensor_spec.position = sensor_params["position"]
+        sensor_spec.orientation = sensor_params["orientation"]
+        sensor_specs.append(sensor_spec)
+
+    # agent configurations
+    agent_cfg = habitat_sim.agent.AgentConfiguration()
+    agent_cfg.sensor_specifications = sensor_specs
+
+    return habitat_sim.Configuration(backend_cfg, [agent_cfg])
+
+
+def make_video_cv2(
```

Comment:
This function should be renamed to be more descriptive. Also, might want to see if we can drop the OpenCV requirement since it's not really necessary anymore. Something like embed_observations_into_video?

Reply:
Do we not need to use cv2 to resize and all that?

Reply:
ImageIO probably has some resize functionality and PIL definitely does.